ispyb/model/integration.py

```diff
@@ -85,3 +91,93 @@ def __str__(self):
     ('detector_distance', 'refinedDetectorDistance'),
     ('timestamp', 'recordTimeStamp'),
 ))
+
+class UnitCell():
+  '''An object representing the parameters of the unit cell I.e unit cell edges and angles
+  '''
+
+  def __init__(self, a,b,c,alpha,beta,gamma):
+    '''Unit cell object
+
+       :param apiid: AutoProcIntegrationID
+       :param db_conn: ISPyB database connection object
+       :return: A unit cell object representing the database entry for
+                the specified AutoProcIntegrationID
+    '''
+    self._a = a
+    self._b = b
+    self._c = c
+    self._alpha = alpha
+    self._beta = beta
+    self._gamma = gamma
+
+   
+  @property
+  def a(self):
+    '''Returns dimension a of unit cell in Angstroms'''
+    return self._a
+
+  @a.setter
+  def a(self, value):
+    '''Sets dimension a of unit cell in Angstroms'''
+    self._a = value
+
+  @property
+  def b(self):
+    '''Returns dimension b of unit cell in Angstroms'''
+    return self._b
+
+  @b.setter
+  def b(self, value):
+    '''Sets dimension b of unit cell in Angstroms'''
+    self._b = value
+
+  @property
+  def c(self):
+    '''Returns dimension c of unit cell in Angstroms'''
+    return self._c
+
+  @c.setter
+  def c(self, value):
+    '''Sets dimension c of unit cell in Angstroms'''
+    self._c = value
+
+  @property
+  def alpha(self):
+    '''Returns angle alpha of unit cell'''
+    return self._alpha
+
+  @alpha.setter
+  def alpha(self, value):
+    '''Sets angle alpha of unit cell'''
+    self._alpha = value
+
+  @property
+  def beta(self):
+    '''Returns angle beta of unit cell'''
+    return self._beta
+
+  @beta.setter
+  def beta(self, value):
+    '''Sets angle beta of unit cell'''
+    self._beta = value
+
+  @property
+  def gamma(self):
+    '''Returns angle gamma of unit cell'''
+    return self._gamma
+
+  @gamma.setter
+  def gamma(self, value):
+    '''Sets angle gamma of unit cell'''
+    self._gamma = value
+ 
+  def __str__(self):
+    '''Returns a pretty-printed object representation.'''
+    text = """  cell a      : {}
+  cell b      : {}
+  cell c      : {}
+  cell alpha  : {}
+  cell beta   : {}
+  cell gamma  : {}""".format(self._a, self._b, self._c, self._alpha, self._beta, self._gamma)
```

Comment:
I would suggest using the same format syntax as in https://github.com/DiamondLightSource/ispyb-api/blob/09dd224f8a861ab0019b223e1dfb7bcbbc5bfab2/ispyb/model/integration.py#L71-L79
which means that you can refer to the parameters at the point where you want them to be shown. That way you don't have a massive ```.format()``` call, and can easily change the order and insert new lines in the future. Also you can use the public parameter names (```.a```) instead of the private variables (```._a```). That way anyone looking at the ```__str__``` method will immediately know how they can access the values.

Reply:
Further information about ```.format()``` at https://pyformat.info/